features/testbot/matching.feature

```diff
@@ -169,3 +169,47 @@ Feature: Basic Map Matching
         When I match I should get
             | trace | matchings | geometry                                   |
             | abd   | abd       | 1,1,1.000089,1,1.000089,1,1.000089,0.99991 |
+
+
+    # Regression test 1 for issue 3176
+    Scenario: Testbot - multiuple segments: properly expose OSM IDs
+        Given the query options
+            | annotations | true    |
+
+        Given the node map
+            """
+            a-1-b--c--d--e--f-2-g
+            """
+
+        And the ways
+            | nodes | oneway |
+            | ab    | no     |
+            | bc    | no     |
+            | cd    | no     |
+            | de    | no     |
+            | ef    | no     |
+            | fg    | no     |
+
+        When I match I should get
+            | trace | OSM IDs       |
+            | 12    | 1,2,3,4,5,6,7 |
```

Comment:
Making our tests more consistent is an important goal, but it does not take precedence over fixing the bug for me.

This works a regression test and will not be flunky since I know the ids will be generated that way. This is more about documenting it to the outside world and ensuring this will always work in the future even if we change the behavior of the support code. These are separate issues from having a working regression test.

For reference consistency is ensured over [this](https://github.com/Project-OSRM/osrm-backend/blob/master/features/step_definitions/data.js#L54-L64). Nodes are processed from left to right, not the code you linked @danpat.